Hard error compiling libc 0.2.36: #![feature(link_cfg)] missing ?

```shell 
 Compiling libc v0.2.36
error[E0658]: is feature gated (see issue #37406)
   --> /cargo/registry/src/github.com-1ecc6299db9ec823/libc-0.2.36/src/unix/mod.rs:279:9
    |
279 |         #[link(name = "c", kind = "static", cfg(target_feature = "crt-static"))]
    |         ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
    |
    = help: add #![feature(link_cfg)] to the crate attributes to enable
```

--- a/appveyor.yml
+++ b/appveyor.yml
@@ -24,4 +24,5 @@ build: false
 
 test_script:
   - cargo test --target %TARGET%
+  - cargo test --no-default-features --target %TARGET%
   - cargo test --manifest-path libc-test/Cargo.toml --target %TARGET%

--- a/ci/run.sh
+++ b/ci/run.sh
@@ -79,5 +79,5 @@ if [ "$TARGET" = "x86_64-unknown-linux-gnux32" ]; then
   opt="--release"
 fi
 
-cargo test $opt --no-default-features --target $TARGET
+cargo test $opt --no-default-features --manifest-path libc-test/Cargo.toml --target $TARGET
 exec cargo test $opt --manifest-path libc-test/Cargo.toml --target $TARGET

--- a/ci/run.sh
+++ b/ci/run.sh
@@ -79,4 +79,5 @@ if [ "$TARGET" = "x86_64-unknown-linux-gnux32" ]; then
   opt="--release"
 fi
 
+exec cargo test $opt --no-default-features --target $TARGET
 exec cargo test $opt --manifest-path libc-test/Cargo.toml --target $TARGET

--- a/ci/run.sh
+++ b/ci/run.sh
@@ -79,5 +79,5 @@ if [ "$TARGET" = "x86_64-unknown-linux-gnux32" ]; then
   opt="--release"
 fi
 
-exec cargo test $opt --no-default-features --target $TARGET
+cargo test $opt --no-default-features --target $TARGET
 exec cargo test $opt --manifest-path libc-test/Cargo.toml --target $TARGET

--- a/libc-test/Cargo.toml
+++ b/libc-test/Cargo.toml
@@ -4,11 +4,16 @@ version = "0.1.0"
 authors = ["Alex Crichton <alex@alexcrichton.com>"]
 build = "build.rs"
 
-[dependencies]
-libc = { path = ".." }
+[dependencies.libc]
+path = ".."
+default-features = false
 
 [build-dependencies]
 ctest = { git = "https://github.com/alexcrichton/ctest" }
+
+[features]
+default = [ "use_std" ]
+use_std = [ "libc/use_std" ]
 
 [[test]]
 name = "main"
@@ -19,3 +24,4 @@ harness = false
 name = "linux-fcntl"
 path = "test/linux_fcntl.rs"
 harness = false
+

--- a/src/unix/mod.rs
+++ b/src/unix/mod.rs
@@ -277,8 +277,11 @@ cfg_if! {
         // cargo build, don't pull in anything extra as the libstd dep
         // already pulls in all libs.
     } else if #[cfg(target_env = "musl")] {
-        #[cfg_attr(feature = "stdbuild", link(name = "c", kind = "static", cfg(target_feature = "crt-static")))]
-        #[cfg_attr(feature = "stdbuild", link(name = "c", cfg(not(target_feature = "crt-static"))))]
+        #[cfg_attr(feature = "stdbuild",
+                   link(name = "c", kind = "static",
+                        cfg(target_feature = "crt-static")))]
+        #[cfg_attr(feature = "stdbuild",
+                   link(name = "c", cfg(not(target_feature = "crt-static"))))]
         extern {}
     } else if #[cfg(target_os = "emscripten")] {
         #[link(name = "c")]

--- a/src/unix/mod.rs
+++ b/src/unix/mod.rs
@@ -286,7 +286,9 @@ cfg_if! {
     } else if #[cfg(target_os = "emscripten")] {
         #[link(name = "c")]
         extern {}
-    } else if #[cfg(all(target_os = "netbsd", cfg_attr(feature = "stdbuild", target_vendor = "rumprun")))] {
+    } else if #[cfg(all(target_os = "netbsd",
+                        cfg_attr(feature = "stdbuild",
+                                 target_vendor = "rumprun")))] {
         // Since we don't use -nodefaultlibs on Rumprun, libc is always pulled
         // in automatically by the linker. We avoid passing it explicitly, as it
         // causes some versions of binutils to crash with an assertion failure.

--- a/src/unix/mod.rs
+++ b/src/unix/mod.rs
@@ -286,12 +286,11 @@ cfg_if! {
     } else if #[cfg(target_os = "emscripten")] {
         #[link(name = "c")]
         extern {}
-    } else if #[cfg(all(target_os = "netbsd",
-                        cfg_attr(feature = "stdbuild",
-                                 target_vendor = "rumprun")))] {
+    } else if #[cfg(all(target_os = "netbsd"))] {
         // Since we don't use -nodefaultlibs on Rumprun, libc is always pulled
         // in automatically by the linker. We avoid passing it explicitly, as it
         // causes some versions of binutils to crash with an assertion failure.
+        #[cfg_attr(feature = "stdbuild", target_vendor = "rumprun")]
         #[link(name = "m")]
         extern {}
     } else if #[cfg(any(target_os = "macos",

--- a/src/unix/mod.rs
+++ b/src/unix/mod.rs
@@ -277,8 +277,8 @@ cfg_if! {
         // cargo build, don't pull in anything extra as the libstd dep
         // already pulls in all libs.
     } else if #[cfg(target_env = "musl")] {
-        #[link(name = "c", kind = "static", cfg(target_feature = "crt-static"))]
-        #[link(name = "c", cfg(not(target_feature = "crt-static")))]
+        #[cfg_attr(feature = "stdbuild", link(name = "c", kind = "static", cfg(target_feature = "crt-static")))]
+        #[cfg_attr(feature = "stdbuild", link(name = "c", cfg(not(target_feature = "crt-static"))))]
         extern {}
     } else if #[cfg(target_os = "emscripten")] {
         #[link(name = "c")]

--- a/src/unix/mod.rs
+++ b/src/unix/mod.rs
@@ -286,7 +286,7 @@ cfg_if! {
     } else if #[cfg(target_os = "emscripten")] {
         #[link(name = "c")]
         extern {}
-    } else if #[cfg(all(target_os = "netbsd", target_vendor = "rumprun"))] {
+    } else if #[cfg(all(target_os = "netbsd", cfg_attr(feature = "stdbuild", target_vendor = "rumprun")))] {
         // Since we don't use -nodefaultlibs on Rumprun, libc is always pulled
         // in automatically by the linker. We avoid passing it explicitly, as it
         // causes some versions of binutils to crash with an assertion failure.

